docs/fold.md

```diff
@@ -0,0 +1,84 @@
+## AIR: Fold instruction
+Here is the `fold` syntax.
+
+`(fold <iterable> <iterator> <instruction> [<last_instruction>])`
+
+Iterable can be: a stream, map, canonicalized stream, canonicalized map or scalar array.
+
+If `last_instruction` is not set, then it's assumed:
+ - `null` in case of `(fold (seq ...`
+ - `never` in case of `(fold (par ...`
+
+```clojure
+(fold iterable iterator
+	(seq
+	  (ap ("key" iterator) %map)
+	  (next iterator)
```

Comment:
Would be nice to provide an example with `next` not as last instruction in `isntruction` also

Reply:
And elaborate on what happens if `next` is **not** called inside `instruction`

Reply:
Good point